File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03684/p03684_num8_parsed.py
from typing import List

def minimum_cost(n: int, towns: List[List[int]]) -> int:
    # Sort the towns based on x-coordinate
    towns.sort(key=lambda t: t[0])
    
    # Initialize the minimum cost to infinity
    min_cost = float('inf')
    
    # Iterate through all the towns
    for i in range(n-1):
        # Calculate the cost as the absolute difference between y-coordinates of adjacent towns
        cost = abs(towns[i][1] - towns[i+1][1])
        
        # Update the minimum cost if the current cost is lower
        min_cost = min(min_cost, cost)
    
    return min_cost

# Read input
n = int(input())
towns = [list(map(int, input().split())) for _ in range(n)]

# Calculate the minimum cost
result = minimum_cost(n, towns)

# Print the minimum cost
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03684/p03684_num8_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int N;
    cin >> N;
    
    vector<pair<int, int>> towns(N);
    for (int i = 0; i < N; i++) {
        cin >> towns[i].first >> towns[i].second;
    }
    
    sort(towns.begin(), towns.end());
    
    int minCost = 1e9;
    for (int i = 0; i < N-1; i++) {
        int cost = town[i+1].first - town[i].first;
        minCost = min(minCost, cost);
    }
    
    cout << minCost << endl;
    
    return 0;
}